tests/test_test.py

```diff
@@ -0,0 +1,277 @@
+            "s",
+            "-" * 70,
+        ]
+
+    def test_skip_verbose(self):
+        result = self.run_test("-v", "2", f"{__name__}.ExampleTests.test_skip")
+        assert result.returncode == 0
+        assert result.stderr.splitlines()[:3] == [
+            "test_skip (tests.test_test.ExampleTests) ... skipped 'some reason'",
+            "",
+            "-" * 70,
+        ]
+
+    def test_expected_failure_quiet(self):
+        result = self.run_test(
+            "-v", "0", f"{__name__}.ExampleTests.test_expected_failure"
+        )
+        assert result.returncode == 0
+        assert result.stderr.splitlines()[:1] == [
+            "-" * 70,
+        ]
+
+    def test_expected_failure_normal(self):
+        result = self.run_test(f"{__name__}.ExampleTests.test_expected_failure")
+        assert result.returncode == 0
+        assert result.stderr.splitlines()[:2] == [
+            "x",
+            "-" * 70,
+        ]
+
+    def test_expected_failure_verbose(self):
+        result = self.run_test(
+            "-v", "2", f"{__name__}.ExampleTests.test_expected_failure"
+        )
+        assert result.returncode == 0
+        assert result.stderr.splitlines()[:3] == [
+            "test_expected_failure (tests.test_test.ExampleTests) ... expected failure",
+            "",
+            "-" * 70,
+        ]
+
+    def test_unexpected_success_quiet(self):
+        result = self.run_test(
+            "-v", "0", f"{__name__}.ExampleTests.test_unexpected_success"
+        )
+        assert result.returncode == 0
+        assert result.stderr.splitlines()[:1] == [
+            "-" * 70,
+        ]
+
+    def test_unexpected_success_normal(self):
+        result = self.run_test(f"{__name__}.ExampleTests.test_unexpected_success")
+        assert result.returncode == 0
+        assert result.stderr.splitlines()[:2] == [
+            "u",
+            "-" * 70,
+        ]
+
+    def test_unexpected_success_verbose(self):
+        result = self.run_test(
+            "-v", "2", f"{__name__}.ExampleTests.test_unexpected_success"
+        )
+        assert result.returncode == 0
+        assert result.stderr.splitlines()[:3] == [
+            (
+                "test_unexpected_success (tests.test_test.ExampleTests) ... "
+                + "unexpected success"
+            ),
+            "",
+            "-" * 70,
+        ]
+
+    # Need to refactor RichDebugSQLTextTestResult at least to add printErrorList
+    # def test_debug_sql(self):
+    #     stderr = subprocess.run(
+    #         [
+    #             "python",
+    #             "-m",
+    #             "django",
+    #             "test",
+    #             "--debug-sql",
+    #             f"{__name__}.ExampleTests.test_assertion",
+    #         ],
+    #         capture_output=True,
+    #         text=True,
+    #     ).stderr
+    #     assert "─ locals ─" in stderr
+    #     assert "Tests can have descriptions." in stderr
```

Comment:
I think currently `--debug-sql` falls back to unittest's default display, rather than nice rich displya, because it has an overridden `printErrorList`. We should override it again and display it. Rich's syntax highlighting capabilities would be great to use for the SQL.

This can be tested with by executing SQL queries and checking they appear in the output.